docs/site/Decorators_inject.md

```diff
@@ -143,6 +143,66 @@ export class HelloController {
 }
 ```
 
+The `setter` function injected has the following signature:
+
+```ts
+/**
+ * The function injected by `@inject.setter(bindingKey)`.
+ */
+export interface Setter<T> {
+  /**
+   * Set the underlying binding to a const value. Returns the `Binding` object.
+   * The usages are:
+   *
+   * @param value Optional value. If not provided, the underlying binding won't
+   * be changed and returned as-is.
+   */
+  (value?: T): Binding<T>;
+}
+```
+
+```ts
+const binding = this.greetingSetter('Greetings!');
+```
+
+or
+
+```ts
+const binding = this.greetingSetter().toDynamicValue(() => 'Greetings!');
```

Comment:
@raymondfeng could you please remind me and other people reviewing this pull request why are we adding this feature? What use cases are you trying to support?

Reply:
Added more description.